source/Octopus.Server.Client/Operations/IRegisterMachineOperationBase.cs

```diff
@@ -53,6 +54,21 @@ public interface IRegisterMachineOperationBase
         /// </summary>
         CommunicationStyle CommunicationStyle { get; set; }
 
+        /// <summary>
+        /// The communication behaviour that Kubernetes Agent will use;
+        /// </summary>
+        public AgentCommunicationStyleResource AgentCommunicationStyle { get; set; }
+
+        /// <summary>
+        /// The default Container that the Kubernetes Agent will use to execute Jobs
+        /// </summary>
+        public string DefaultJobExecutionContainer { get; set; }
+
+        /// <summary>
+        /// The default Container Feed that the Kubernetes Agent will use to retrieve Job Execution Containers.
+        /// </summary>
+        public string DefaultJobExecutionContainerFeed { get; set; }
```

Comment:
Are we able to leave these out until we know the structure we want for these? I can see a scenario where we don't include these defaults for an MVP, but do them after

Reply:
Ah yep fair enough, good call 👍